plugins/graph_algorithm/src/algorithms/components.cpp

```diff
@@ -0,0 +1,73 @@
+#include "graph_algorithm/algorithms/components.h"
+
+#include "graph_algorithm/netlist_graph.h"
+
+#include <map>
+
+namespace hal
+{
+    namespace graph_algorithm
+    {
+        Result<std::vector<std::vector<u32>>> get_connected_components(NetlistGraph* graph, bool strong, u32 min_size)
+        {
+            if (graph == nullptr)
+            {
+                return ERR("graph is a nullptr");
+            }
+
+            igraph_vector_int_t membership, csize;
+            igraph_integer_t number_of_clusters;
+            auto err = igraph_vector_int_init(&membership, 0);
+            if (err != IGRAPH_SUCCESS)
+            {
+                igraph_vector_int_destroy(&membership);
+                return ERR(igraph_strerror(err));
+            }
+
+            err = igraph_vector_int_init(&csize, 0);
+            if (err != IGRAPH_SUCCESS)
+            {
+                igraph_vector_int_destroy(&membership);
+                igraph_vector_int_destroy(&csize);
+                return ERR(igraph_strerror(err));
+            }
```

Comment:
```suggestion
            err = igraph_vector_int_init(&csize, 0);
            if (err != IGRAPH_SUCCESS)
            {
                igraph_vector_int_destroy(&membership);
                return ERR(igraph_strerror(err));
            }
```

Same comment about destroying data structure whose initialization has failed (here and many other places).